contracts/test/integration/IntegrationTestMakerRouter.t.sol

```diff
@@ -0,0 +1,357 @@
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        makerRouter.swapAllFeiForUsdcAndDai(
+            address(this),
+            address(this),
+            ratioUSDC
+        );
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        assertEq(
+            usdc.balanceOf(address(this)),
+            usdcAmount / USDC_SCALING_FACTOR
+        );
+        assertEq(dai.balanceOf(address(this)), minDaiAmountOut - usdcAmount);
+    }
+
+    function testSwapForDaiRevertWithoutPermissions() public {
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        vm.expectRevert("UNAUTHORIZED");
+        makerRouter.swapFeiForDai(mintAmount, minDaiAmountOut, address(this));
+    }
+
+    function testSwapAllForDaiRevertWithoutPermissions() public {
+        vm.expectRevert("UNAUTHORIZED");
+        makerRouter.swapAllFeiForDai(address(this));
+    }
+
+    function testSwapForUsdcRevertWithoutPermissions() public {
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        vm.expectRevert("UNAUTHORIZED");
+        makerRouter.swapFeiForUsdc(mintAmount, minDaiAmountOut, address(this));
+    }
+
+    function testSwapAllForUsdcRevertWithoutPermissions() public {
+        vm.expectRevert("UNAUTHORIZED");
+        makerRouter.swapAllFeiForUsdc(address(this));
+    }
+
+    function testSwapForUsdcAndDaiRevertWithoutPermissions() public {
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        vm.expectRevert("UNAUTHORIZED");
+        makerRouter.swapFeiForUsdcAndDai(
+            mintAmount,
+            minDaiAmountOut,
+            address(this),
+            address(this),
+            5000
+        );
+    }
+
+    function testSwapAllForUsdcAndDaiRevertWithoutPermissions() public {
+        vm.expectRevert("UNAUTHORIZED");
+        makerRouter.swapAllFeiForUsdcAndDai(address(this), address(this), 5000);
+    }
+
+    function testInvalidAboveUSDCRatio() public {
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+
+        vm.expectRevert("MakerRouter: Invalid USDC Ratio");
+        makerRouter.swapFeiForUsdcAndDai(
+            mintAmount,
+            minDaiAmountOut,
+            address(this),
+            address(this),
+            10000
+        );
+    }
+
+    function testInvalidBelowUSDCRatio() public {
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+
+        vm.expectRevert("MakerRouter: Invalid USDC Ratio");
+        makerRouter.swapFeiForUsdcAndDai(
+            mintAmount,
+            minDaiAmountOut,
+            address(this),
+            address(this),
+            0
+        );
+    }
```

Comment:
Let's add a test here where we hit the "MakerRouter: Not enough USDC out" require statement.